types/api.d.ts

```diff
@@ -0,0 +1,49 @@
+import {Uri} from 'vscode';
+
+declare interface BasToolkit {
+    getExtensionAPI: <T>(extensionId: string) => Promise<T>;
+    getAction: (actionId: string) => IAction | undefined;
+    getParameter: (parameterName: string) => Promise<string | undefined>;
+
+    actions: {
+        performAction: (action: IAction, options?: any) => Thenable<void>;
+        ExecuteAction: { new(): IExecuteAction };
+        SnippetAction: { new(): ISnippetAction };
+        CommandAction: { new(): ICommandAction };
+        FileAction: { new(): IFileAction };
+    };
+}
+
+declare const bas: BasToolkit;
+
+declare type ActionType = "EXECUTE" | "COMMAND" | "TASK" | "FILE" | "SNIPPET"
+
+declare type CommandActionParams = any[];
+declare type ExecuteActionParams = any[];
+declare type SnippetActionParams = Record<string,any>
+
+interface IAction {
+    id?: string;
+    actionType: ActionType | undefined;
+}
+
+interface IExecuteAction extends IAction {
+    executeAction: (params?: ExecuteActionParams) => Thenable<any>;
```

Comment:
I think we discussed overriding the `actionType` in each sub-interface with a more specific value